webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/services/TechniqueRepository.scala

```diff
@@ -122,14 +122,20 @@ trait TechniqueRepository {
 
   def getParentTechniqueCategory_forTechnique(id: TechniqueId): Box[TechniqueCategory]
 
-  final def getTechniqueCategoriesBreadCrump(id: TechniqueId): Box[Seq[TechniqueCategory]] = {
+  final
+  def getTechniqueCategoriesBreadCrump(id: TechniqueId): Box[Seq[TechniqueCategory]] = {
     for {
       cat <- getParentTechniqueCategory_forTechnique(id)
       path <- sequence(cat.id.getIdPathFromRoot) { currentCatId =>
-        getTechniqueCategory(currentCatId) ?~! "'%s' category was not found but should be a parent of '%s'".format(currentCatId, cat.id)
+        getTechniqueCategory(currentCatId) ?~! "'%s' category was not found but should be a parent of '%s'".format(
```

Comment:
You could have switch to a string interpolation ! and more importantly you should use currentCatId.,name and cat.id.name